library/src-bootstrapped/scala/quoted/util/QuoteUtils.scala

```diff
@@ -0,0 +1,24 @@
+package scala.quoted
+package util
+
+/** An abstraction for variable definition to use in a quoted program.
+ *  It decouples the operations of get and update, if needed to be spliced separately.
+ */
+sealed trait Var[T] {
+  def get given QuoteContext: Expr[T]
+  def update(x: Expr[T]) given QuoteContext: Expr[Unit]
+}
+
+object Var {
+  def apply[T: Type, U: Type](init: Expr[T])(body: Var[T] => Expr[U]) given QuoteContext: Expr[U] = '{
```

Comment:
Add some documentation with an example of how to use this. Maybe adapt this one https://github.com/nicolasstucki/quoted-util/blob/master/src/main/scala/quoted/util/Let.scala#L67